api/src/bus/lib/use-bus.ts

```diff
@@ -0,0 +1,24 @@
+import { createBus, type Bus } from '@directus/memory';
+import { redisConfigAvailable, useRedis } from '../../redis/index.js';
+
+export const _cache: { bus: Bus | undefined } = {
+	bus: undefined,
+};
+
+/**
+ * Returns globally shared message bus. If Redis is available, will use a redis-driven pub/sub bus.
+ * Otherwise will default to a local-only bus.
+ */
+export const useBus = () => {
```

Comment:
Now that we call this from different places, which _might_ happen around the same time, it could result in multiple different instances being created and used.

The chance is very small, but still worries me, because it would be **bad**.
Will probably be the same for future refactoring of similar kind.

So I think we need a Promise here.

Reply:
The function is synchronous and due to JS' single-threadedness, you won't have the risk of having multiple redis instances. By making it a promise, you actually expose the risk of multiple things running it in parallel! 🙂

Reply:
Of course, don't know what I was thinking 😅🙏

![byebye](https://github.com/directus/directus/assets/5363448/30fdd5b5-bf81-4a18-b8d4-ae587cde4659)